cub/test/test_device_scan.cu

```diff
@@ -1,1275 +0,0 @@
-void TestSize(
-    int     num_items,
-    OutputT identity,
-    OutputT initial_value)
-{
-    if (num_items < 0)
-    {
-        TestOp<InputT>(0,        identity, initial_value);
-        TestOp<InputT>(1,        identity, initial_value);
-        TestOp<InputT>(100,      identity, initial_value);
-        TestOp<InputT>(10000,    identity, initial_value);
-        TestOp<InputT>(1000000,  identity, initial_value);
-    }
-    else
-    {
-        TestOp<InputT>(num_items, identity, initial_value);
-    }
-}
-
-class CustomInputT
-{
-  char m_val{};
-
-public:
-  __host__ __device__ explicit CustomInputT(char val)
-      : m_val(val)
-  {}
-
-  __host__ __device__ int get() const { return static_cast<int>(m_val); }
-};
-
-class CustomAccumulatorT
-{
-  int m_val{0};
-  int m_magic_value{42};
-
-  __host__ __device__ CustomAccumulatorT(int val)
-      : m_val(val)
-  {}
-
-public:
-  __host__ __device__ CustomAccumulatorT()
-  {}
-
-  __host__ __device__ CustomAccumulatorT(const CustomAccumulatorT &in)
-    : m_val(in.is_valid() * in.get())
-    , m_magic_value(in.is_valid() * 42)
-  {}
-
-  __host__ __device__ CustomAccumulatorT(const CustomInputT &in)
-    : m_val(in.get())
-    , m_magic_value(42)
-  {}
-
-  __host__ __device__ void operator=(const CustomInputT &in)
-  {
-    if (this->is_valid())
-    {
-      m_val = in.get();
-    }
-  }
-
-  __host__ __device__ void operator=(const CustomAccumulatorT &in)
-  {
-    if (this->is_valid() && in.is_valid())
-    {
-      m_val = in.get();
-    }
-  }
-
-  __host__ __device__ CustomAccumulatorT 
-  operator+(const CustomInputT &in) const
-  {
-    const int multiplier = this->is_valid();
-    return {(m_val + in.get()) * multiplier};
-  }
-
-  __host__ __device__ CustomAccumulatorT
-  operator+(const CustomAccumulatorT &in) const
-  {
-    const int multiplier = this->is_valid() && in.is_valid();
-    return {(m_val + in.get()) * multiplier};
-  }
-
-  __host__ __device__ int get() const { return m_val; }
-
-  __host__ __device__ bool is_valid() const { return m_magic_value == 42; }
-};
-
-class CustomOutputT
-{
-  int *m_d_ok_count{};
-  int m_expected{};
-
-public:
-  __host__ __device__ CustomOutputT(int *d_ok_count, int expected)
-      : m_d_ok_count(d_ok_count)
-      , m_expected(expected)
-  {}
-
-  __device__ void operator=(const CustomAccumulatorT &accum) const
-  {
-    const int ok = accum.is_valid() && (accum.get() == m_expected);
-    atomicAdd(m_d_ok_count, ok);
-  }
-};
-
-__global__ void InitializeTestAccumulatorTypes(int num_items,
-                                               int *d_ok_count,
-                                               CustomInputT *d_in,
-                                               CustomOutputT *d_out)
-{
-  const int idx = static_cast<int>(threadIdx.x + blockIdx.x * blockDim.x);
-
-  if (idx < num_items)
-  {
-    d_in[idx] = CustomInputT(1);
-    d_out[idx] = CustomOutputT{d_ok_count, idx};
-  }
-}
-
-void TestAccumulatorTypes()
```

Comment:
Is there an alternative to `TestAccumulatorTypes` in the new tests?

Reply:
Thanks! I've added a test that resembles `TestAccumulatorTypes` to `catch2_test_device_scan_iterators.cu`